net/ceph/osd_client.c

```diff
@@ -3391,6 +3463,116 @@ int ceph_osdc_notify_ack(struct ceph_osd_client *osdc,
 }
 EXPORT_SYMBOL(ceph_osdc_notify_ack);
 
+static int osd_req_op_notify_init(struct ceph_osd_request *req, int which,
+				  u64 cookie, u32 prot_ver, u32 timeout,
+				  void *payload, size_t payload_len)
+{
+	struct ceph_osd_req_op *op;
+	struct ceph_pagelist *pl;
+	int ret;
+
+	op = _osd_req_op_init(req, which, CEPH_OSD_OP_NOTIFY, 0);
+	op->notify.cookie = cookie;
+
+	pl = kmalloc(sizeof(*pl), GFP_NOIO);
+	if (!pl)
+		return -ENOMEM;
+
+	ceph_pagelist_init(pl);
+	ret = ceph_pagelist_encode_32(pl, 1); /* prot_ver */
+	ret |= ceph_pagelist_encode_32(pl, timeout);
+	ret |= ceph_pagelist_encode_32(pl, payload_len);
```

Comment:
The payload is encoded with an encoding start block. I think the correct encoding is:
u32 proto_ver (=1)
u32 timeout
u32 block_len (= payload_len + CEPH_ENCODING_START_BLOCK_LEN)
u8 cur_ver (=1)
u8 compat_ver (=1)
u32 len (= payload_len)
payload

cf. https://github.com/fullerdj/ceph-client/commit/2950b49b6b2c3d42a10612dd7aba72836655f80c